src/Microsoft.ML.StandardLearners/FactorizationMachine/FieldAwareFactorizationMachineModelParameters.cs

```diff
@@ -237,43 +250,33 @@ internal void CopyLatentWeightsTo(AlignedArray latentWeights)
             latentWeights.CopyFrom(_latentWeightsAligned);
         }
 
-        /// <summary>
-        /// Get the number of fields. It's the symbol `m` in the doc: https://github.com/wschin/fast-ffm/blob/master/fast-ffm.pdf
-        /// </summary>
-        public int GetFieldCount() => FieldCount;
-
-        /// <summary>
-        /// Get the number of features. It's the symbol `n` in the doc: https://github.com/wschin/fast-ffm/blob/master/fast-ffm.pdf
-        /// </summary>
-        public int GetFeatureCount() => FeatureCount;
-
-        /// <summary>
-        /// Get the latent dimension. It's the tlngth of `v_{j, f}` in the doc: https://github.com/wschin/fast-ffm/blob/master/fast-ffm.pdf
-        /// </summary>
-        public int GetLatentDim() => LatentDim;
-
         /// <summary>
         /// The linear coefficients of the features. It's the symbol `w` in the doc: https://github.com/wschin/fast-ffm/blob/master/fast-ffm.pdf
         /// </summary>
-        public float[] GetLinearWeights() => _linearWeights;
+        public float[] GetLinearWeights()
```

Comment:
>float[] [](start = 15, length = 7)

I would reach Tom and double check, but I think we don't want expose arrays. We want expose immutable collections, either `ImmutableArray` or `IReadOnlyList` not sure which we prefer.
If you expose array, that 
a) force you to copy data.
b) give user idea what he can change it. (which is not true)

Reply:
In tree's public interface, we expose `IReadOnlyList` for all array attributes.

Reply:
What about `ReadOnlyCollection`? Any idea @Tom? what we are following?

---
In reply to: [260523116](https://github.com/dotnet/machinelearning/pull/2730#discussion_r260523116) [](ancestors = 260523116)

Reply:
OK, I follow what @wschin suggest. `IReadOnlyList`